presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/PlanMatchPattern.java

```diff
@@ -247,8 +249,57 @@ private String indentString(int indent)
         return Strings.repeat("    ", indent);
     }
 
-    private static class AnySymbolReference
-            extends SymbolReference
+    public static SymbolStem symbolStem(String stem)
+    {
+        return new SymbolStem(stem);
+    }
+
+    public static class SymbolStem extends Symbol
+    {
+        private final String stem;
+
+        SymbolStem(String stem)
+        {
+            // For useful output when we print out failure messages.
+            super(stem + "_*");
+            this.stem = stem;
+        }
+
+        @Override
+        public boolean equals(Object o)
+        {
+            if (this == o) {
+                return true;
+            }
+
+            if (o == null) {
+                return false;
+            }
+
+            if (Symbol.class.equals(o.getClass())) {
+                Symbol other = (Symbol) o;
+                String name = other.getName();
+                int endIndex = name.lastIndexOf('_');
+                String otherStem = endIndex == -1 ? name : name.substring(0, endIndex);
+                return stem.equals(otherStem);
+            }
+
+            if (!SymbolStem.class.equals(o.getClass())) {
+                return false;
+            }
+
+            SymbolStem other = (SymbolStem) o;
+            return stem.equals(other.stem);
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return stem.hashCode();
```

Comment:
To me contract between `hashCode` and `equals` is broken here. `hashCode` for StemSymbol returns the the hash code of the stem (name_) while `Symbol#hashCode` returns for hash code of the symbol name (name_13). They will most likely diiffer to each other, while the `equals` would return `true`.

Please correct me if I am wrong. If it is true, please throw an exception here instead.